test/uk/gov/hmrc/mtdsatestsupportapi/controllers/DeleteTestBusinessControllerSpec.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mtdsatestsupportapi.controllers
+
+import api.controllers.{ControllerBaseSpec, ControllerTestRunner}
+import api.models.domain.Nino
+import api.models.errors.{BusinessIdFormatError, ErrorWrapper, InternalError, NinoFormatError, NotFoundError}
+import api.models.outcomes.ResponseWrapper
+import play.api.http.HeaderNames
+import play.api.mvc.Result
+import uk.gov.hmrc.mtdsatestsupportapi.mocks.requestParsers.MockDeleteTestBusinessRequestParser
+import uk.gov.hmrc.mtdsatestsupportapi.mocks.services.MockDeleteTestBusinessService
+import uk.gov.hmrc.mtdsatestsupportapi.models.request.deleteTestBusiness.{DeleteTestBusinessRawData, DeleteTestBusinessRequest}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class DeleteTestBusinessControllerSpec
+    extends ControllerBaseSpec
+    with ControllerTestRunner
+    with MockDeleteTestBusinessRequestParser
+    with MockDeleteTestBusinessService {
+
+  trait Test extends ControllerTest {
+    val nino           = "AA123456A"
+    val businessId     = "XAIS12345678910"
+    val vendorClientId = "some_id"
+
+    val rawData: DeleteTestBusinessRawData     = DeleteTestBusinessRawData(nino, businessId)
+    val requestData: DeleteTestBusinessRequest = DeleteTestBusinessRequest(Nino(nino), businessId)
+
+    val controller = new DeleteTestBusinessController(
+      cc = cc,
+      authService = mockEnrolmentsAuthService,
+      parser = mockParser,
+      service = mockService,
+      idGenerator = mockIdGenerator
+    )
+
+  }
+
+  "DeleteTestBusinessController" must {
+    "return 500 Internal server error" when {
+      "the X-Client-Id header is missing" in new Test {
+        override protected def callController(): Future[Result] =
+          controller.handleRequest(nino, businessId)(fakeRequestWithHeaders(HeaderNames.AUTHORIZATION -> "Bearer Token"))
+
+        val result: Future[Result] = callController()
+
+        status(result) shouldBe InternalError.httpStatus
+      }
+    }
+    "return the correct error" when {
+      "the parser validation rejects the nino" in new Test {
+        override protected def callController(): Future[Result] =
+          controller.handleRequest(nino, businessId)(
+            fakeRequestWithHeaders(HeaderNames.AUTHORIZATION -> "Bearer Token", "X-Client-Id" -> vendorClientId))
```

Comment:
To make the ITs a little realistic, we could change the nino here to something obviously in a wrong format e.g. `val badNino = "bad_nino"`

Reply:
I used to do this, but then it was pointed out on another PR (see [here](https://github.com/hmrc/mtd-sa-test-support-api/pull/46#discussion_r1309957910)) that since we're mocking the parser it doesn't actually make any difference what the nino is. In which case it's arguably misleading to change the nino value as if it did make a difference.